web-api/workflow-terraform/migration/main/lambdas/migrations/0010-add-lead-docket-number-to-case-deadline.test.ts

```diff
@@ -0,0 +1,101 @@
+  trialSessionId: '7805d1ab-18d0-43ec-bafb-654e83405410',
+};
+
+const MOCK_CASE_RECORD = {
+  associatedJudge: 'Colvin',
+  docketNumber: '101-18',
+  docketNumberWithSuffix: '101-18',
+  entityName: 'Case',
+  pk: 'case|101-18',
+  sk: 'case|101-18',
+  status: CASE_STATUS_TYPES.calendared,
+  trialDate: '2020-03-01T00:00:00.000Z',
+  trialLocation: 'Washington, District of Columbia',
+  trialSessionId: '7805d1ab-18d0-43ec-bafb-654e83405410',
+};
+
+describe('migrateItems', () => {
+  it('should return and not modify records that do not have a leadDocketNUmber', async () => {
+    (queryFullCase as jest.Mock).mockResolvedValue({
+      pk: `case|${MOCK_CASE_RECORD.docketNumber}`,
+      sk: `case|${MOCK_CASE_RECORD.docketNumber}`,
+    });
+    (aggregateCaseItems as jest.Mock).mockReturnValue(MOCK_CASE_RECORD);
+
+    const mockNonConsolidatedCaseDeadlineItem = [
+      {
+        associatedJudge: MOCK_CASE_RECORD.associatedJudge,
+        caseDeadLineId: '97a214a0-437d-461b-80a9-1cfd3d669690',
+        createdAt: '2023-06-06T18:55:57.361Z',
+        deadlineDate: '2023-06-10T04:00:00.000Z',
+        description: 'test',
+        docketNumber: MOCK_CASE_RECORD.docketNumber,
+        entityName: 'CaseDeadline',
+        pk: 'case-deadline|97a214a0-437d-461b-80a9-1cfd3d669690',
+        sk: 'case-deadline|97a214a0-437d-461b-80a9-1cfd3d669690',
+        sortableDocketNumber: 2018000101,
+      },
+    ];
+
+    const results = await migrateItems(
+      mockNonConsolidatedCaseDeadlineItem,
+      documentClientMock,
+    );
+
+    expect(results).toEqual(mockNonConsolidatedCaseDeadlineItem);
+  });
+
+  it('should modify CaseDeadline when the case has a leadDocketNumber, adding the case\'s property "leadDocketNUmber" to the CaseDeadline', async () => {
+    (queryFullCase as jest.Mock).mockResolvedValue({
+      pk: `case|${MOCK_CONSOLIDATED_CASE_RECORD.docketNumber}`,
+      sk: `case|${MOCK_CONSOLIDATED_CASE_RECORD.docketNumber}`,
+    });
+    (aggregateCaseItems as jest.Mock).mockReturnValue(
+      MOCK_CONSOLIDATED_CASE_RECORD,
+    );
+
+    const mockConsolidatedCaseDeadlineItem = [
+      {
+        associatedJudge: MOCK_CONSOLIDATED_CASE_RECORD.associatedJudge,
+        caseDeadLineId: '97a214a0-437d-461b-80a9-1cfd3d669690',
+        createdAt: '2023-06-06T18:55:57.361Z',
+        deadlineDate: '2023-06-10T04:00:00.000Z',
+        description: 'test',
+        docketNumber: MOCK_CONSOLIDATED_CASE_RECORD.docketNumber,
+        entityName: 'CaseDeadline',
+        leadDocketNumber: MOCK_CONSOLIDATED_CASE_RECORD.leadDocketNumber,
+        pk: 'case-deadline|97a214a0-437d-461b-80a9-1cfd3d669690',
+        sk: 'case-deadline|97a214a0-437d-461b-80a9-1cfd3d669690',
+        sortableDocketNumber: 2018000101,
+      },
+    ];
+
+    const results = await migrateItems(
+      mockConsolidatedCaseDeadlineItem,
+      documentClientMock,
+    );
+
+    expect(results).toEqual(mockConsolidatedCaseDeadlineItem);
```

Comment:
Shouldn't `mockConsolidatedCaseDeadlineItem` NOT have `leadDocketNumber`? 

And then we would expect the results to have `leadDocketNumber` because that's what `migrateItems(...)` does?

Reply:
I modify it by removing the leadDocketNumber from the original, then created a clone of the original and added the leadDocketNUmber to compare to